src/main/java/com/syntaxphoenix/spigot/smoothtimber/listener/BlockBreakListener.java

```diff
@@ -82,76 +77,18 @@ public void onBlockBreak(final BlockBreakEvent event) {
-                                if (player.getGameMode() != GameMode.CREATIVE && change.removeDurabilityFromItem(tool) == null) {
-                                    break;
-                                }
-                                event.add(wood);
-                                final int amount = maxItems <= 1 ? maxItems : generateAmount(maxItems);
-                                if (animated) {
-                                    final Entity entity = change.toFallingBlock(block);
-                                    entity.setMetadata("STAnimate", new FixedMetadataValue(plugin, amount));
-                                    if (collect) {
-                                        entity.setMetadata("STCollect",
-                                            new FixedMetadataValue(plugin, player.getUniqueId().toString()));
-                                    }
-                                    continue;
-                                }
-                                if (collect) {
-                                    final ItemStack stack = change.getItemFromBlock(block);
-                                    stack.setAmount(amount);
-                                    player.getInventory().addItem(stack);
-                                    continue;
-                                }
-                                change.dropItemByBlock(block, amount);
-                            }
-                        }
-                        Bukkit.getScheduler().runTaskAsynchronously(PluginUtils.MAIN, () -> SmoothTimber.triggerFallEvent(event));
-                    }
+						final int maxItems = CutterConfig.ENABLE_LUCK ? change.getMaxDropCount(tool) : 1;
+						final ArrayList<Location> woodBlocks = new ArrayList<>();
 
-                    private int generateAmount(final int max) {
-                        int drop = 1;
-                        final float more = 1f / (max + 1);
-                        float previous = more * 2f;
-                        float next = more * 3f;
-                        final float chance = generator.nextFloat() * (float) CutterConfig.LUCK_MULTIPLIER;
-                        while (true) {
-                            if (previous < chance && chance > next) {
-                                drop++;
-                                previous = next;
-                                next += more;
-                            } else {
-                                if (previous < chance && chance < next) {
-                                    drop++;
-                                }
-                                break;
-                            }
-                        }
-                        return Math.min(drop, 64);
-                    }
-
-                });
-            });
+						BlockBreakTask blockBreakTask = new BlockBreakTask(player,location,change,tool,woodBlocks,maxItems,generator);
```

Comment:
Also formatting again.
It would be great if you could put a space after each `,` as well